### DISCUSSION TITLE
Quorum Node

### DISCUSSION THREAD

**Discussion by git-varun** [ConsenSys/quorum#1442] [Q&A] [answered]
In case we are to migrate an existing Quorum node setup from a set of servers (VMs) to completely another set of servers (VMs) with obvious changes in Public IPs, can we retain the private data?
If so, what would be the process for that? Specific step details would be helpful.
Thanks in advance.

**Comment by Julian332** [comment]
I just offer an idea： you can add  new set of servers to the old network , and   disable  the old servers  after synchronization is complete.
you can find the Tutorials about how  to add a new  node in offical site of quorum .

**Reply by git-varun** [reply]
we need to move from one subscription to another, so adding new servers in the same VNET is impossible. Public IPs will be changed in our case. Do you know if this will impact retrieving the private data?

**Comment by baptiste-b-pegasys** [comment, marked as answer]
Hello, the private data is stored in the database of Tessera.
I haven't done this exercise, but the plan could be:
1. prepare the new node (setup, database ready, modified config files: IP modified, but same public/private keys to communicate securely)
2. turn down node, the goquorum instance, even tessera but not the database
3. backup the database from the old node in order to restore it in the new node
4. launch tessera, then go quorum
5. go quorum will resync public data from other nodes, with the same enode, public and private keys, you should be able to read the private data from tessera

Moreover, depending on the permissing you are using, you should add the new node to all others (except the old node, that should be removed at the same time)

I hope this make sense.

**Reply by Purbaja** [reply]
Hi @baptiste-b-pegasys , Thanks for the steps. 

Just wanted to know as the public ip of the VM will change, before stoping the old node should we perform raft.removePeer(oldQuorumNodeId) ? and then after migration start the quorum node from new VM with raft.addPeer(oldQuorumNodeId) ? or we have to assign a new nodeId to the Quorum node while starting up from new VM?

**Reply by baptiste-b-pegasys** [reply]
good point, I guess you are right, you should do that in order to have the correct state on raft.

If I were you, I would practice the plan before doing anything in prod.
